sky/clouds/aws.py

```diff
@@ -480,34 +481,54 @@ def get_image_size(cls, image_id: str, region: Optional[str]) -> float:
-                'VolumeSize']
-        except (aws.botocore_exceptions().NoCredentialsError,
-                aws.botocore_exceptions().ProfileNotFound) as e:
-            logger.debug(
-                f'Failed to get image size for {image_id} in region {region}: {e}'
-            )
-            # Fallback to default image size if no credentials are available.
-            # The credentials issue will be caught when actually provisioning
-            # the instance and appropriate errors will be raised there.
-            return DEFAULT_AMI_GB
-        except aws.botocore_exceptions().ClientError as e:
-            logger.debug(
-                f'Failed to get image size for {image_id} in region {region}: {e}'
-            )
-            with ux_utils.print_exception_no_traceback():
-                raise ValueError(image_not_found_message) from None
+        max_retries = 3
+        for iteration in range(1, max_retries + 1):
+            try:
+                client = aws.client('ec2', region_name=region)
+                image_info = client.describe_images(ImageIds=[image_id]).get(
+                    'Images', [])
+                if not image_info:
+                    with ux_utils.print_exception_no_traceback():
+                        raise ValueError(image_not_found_message)
+                image_size = image_info[0]['BlockDeviceMappings'][0]['Ebs'][
+                    'VolumeSize']
+                break
+            except (aws.botocore_exceptions().NoCredentialsError,
+                    aws.botocore_exceptions().ProfileNotFound) as e:
+                logger.debug(
+                    f'Failed to get image size for {image_id} in region {region}: {e}'
+                )
+                # Fallback to default image size if no credentials are available.
+                # The credentials issue will be caught when actually provisioning
+                # the instance and appropriate errors will be raised there.
+                return DEFAULT_AMI_GB
+            except aws.botocore_exceptions().ClientError as e:
+                logger.debug(
+                    f'Failed to get image size for {image_id} in region {region}: {e}'
+                )
+                if iteration == max_retries:
+                    with ux_utils.print_exception_no_traceback():
+                        raise ValueError(image_not_found_message) from None
+            # linear backoff starting from 0.5 seconds
+            time.sleep(iteration * 0.5)
+        # cache the result for a day.
+        # AMIs are immutable, so we can cache the result for a long time.
+        # While AMIs can be deleted, if the AMI is deleted before cache expiration,
+        # the actual VM launch still fails.
+        day_in_seconds = 60 * 60 * 24  # 1 day, 60s * 60m * 24h
+        kv_cache.add_or_update_cache_entry(kv_cache_key, str(image_size),
+                                           time.time() + day_in_seconds)
```

Comment:
Considering that this could be a remote DB call which could fail transiently as well, should we wrap it in a try block and just log the error and proceed instead of raising, since it isn't a critical error and we can cache it next time.